tests: run-params issue

### Issue and Steps to Reproduce

Testing on Debian buster/sid on aarch64 (DragonBoard 410c with latest [build 359](http://releases.linaro.org/96boards/dragonboard410c/linaro/debian/18.01/))

UPDATE: Also Alpine Linux on `x86_64` see [later comment](https://github.com/ElementsProject/lightning/issues/1668#issuecomment-403707482)

I can give root access to this box, just send me your public SSH key via email.

```
linaro@linaro-developer:~/lightning-auto-test/lightning$ git rev-parse HEAD
13796f0d0f256e15ac20f35ad9e23944965f09cf
linaro@linaro-developer:~/lightning-auto-test/lightning$ make check
bitcoin/test/run-tx-encode > /dev/null
common/test/run-sphinx > /dev/null
common/test/run-json > /dev/null
do_json_tok_bitcoin_amount("0.00000001", 1): ok
do_json_tok_bitcoin_amount("0.00000007", 7): ok
do_json_tok_bitcoin_amount("0.00000008", 8): ok
do_json_tok_bitcoin_amount("0.00000010", 10): ok
do_json_tok_bitcoin_amount("0.12345678", 12345678): ok
do_json_tok_bitcoin_amount("0.01234567", 1234567): ok
do_json_tok_bitcoin_amount("123.45678900", 12345678900): ok
common/test/run-bolt11 > /dev/null
common/test/run-features > /dev/null
common/test/run-ip_port_parsing > /dev/null
wire/test/run-peer-wire > /dev/null
wallet/test/run-db > /dev/null
wallet/test/run-wallet > /dev/null
gossipd/test/run-find_route-specific > /dev/null
gossipd/test/run-initiator-success > /dev/null
gossipd/test/run-find_route > /dev/null
gossipd/test/run-bench-find_route > /dev/null
gossipd/test/run-responder-success > /dev/null
channeld/test/run-full_channel > /dev/null
onchaind/test/run-grind_feerate > /dev/null
lightningd/test/run-commit_tx > /dev/null
lightningd/test/run-key_derive > /dev/null
lightningd/test/run-params > /dev/null
Aborted
Makefile:378: recipe for target 'unittest/lightningd/test/run-params' failed
make: *** [unittest/lightningd/test/run-params] Error 134
```

And the strace of `run-params` is here:

```
linaro@linaro-developer:~/lightning-auto-test/lightning$ strace lightningd/test/run-params
execve("lightningd/test/run-params", ["lightningd/test/run-params"], 0xfffff85b00e0 /* 21 vars */) = 0
brk(NULL)                               = 0xaaaadde35000
faccessat(AT_FDCWD, "/etc/ld.so.nohwcap", F_OK) = -1 ENOENT (No such file or directory)
faccessat(AT_FDCWD, "/etc/ld.so.preload", R_OK) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/etc/ld.so.cache", O_RDONLY|O_CLOEXEC) = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=55765, ...}) = 0
mmap(NULL, 55765, PROT_READ, MAP_PRIVATE, 3, 0) = 0xffffbea15000
close(3)                                = 0
faccessat(AT_FDCWD, "/etc/ld.so.nohwcap", F_OK) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/lib/aarch64-linux-gnu/libm.so.6", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0\267\0\1\0\0\0@\257\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0644, st_size=694360, ...}) = 0
mmap(NULL, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0xffffbe9f6000
mmap(NULL, 757880, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xffffbe93c000
mprotect(0xffffbe9e5000, 61440, PROT_NONE) = 0
mmap(0xffffbe9f4000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0xa8000) = 0xffffbe9f4000
close(3)                                = 0
faccessat(AT_FDCWD, "/etc/ld.so.nohwcap", F_OK) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/lib/aarch64-linux-gnu/libgmp.so.10", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0\267\0\1\0\0\0\240\301\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0644, st_size=463440, ...}) = 0
mmap(NULL, 527144, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xffffbe8bb000
mprotect(0xffffbe92b000, 61440, PROT_NONE) = 0
mmap(0xffffbe93a000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x6f000) = 0xffffbe93a000
close(3)                                = 0
faccessat(AT_FDCWD, "/etc/ld.so.nohwcap", F_OK) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/lib/aarch64-linux-gnu/libsqlite3.so.0", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0\267\0\1\0\0\0@\367\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0644, st_size=997008, ...}) = 0
mmap(NULL, 1062544, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xffffbe7b7000
mprotect(0xffffbe8a5000, 65536, PROT_NONE) = 0
mmap(0xffffbe8b5000, 20480, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0xee000) = 0xffffbe8b5000
mmap(0xffffbe8ba000, 1680, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0xffffbe8ba000
close(3)                                = 0
faccessat(AT_FDCWD, "/etc/ld.so.nohwcap", F_OK) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/lib/aarch64-linux-gnu/libz.so.1", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0\267\0\1\0\0\0`$\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0644, st_size=117032, ...}) = 0
mmap(NULL, 180616, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xffffbe78a000
mprotect(0xffffbe7a6000, 61440, PROT_NONE) = 0
mmap(0xffffbe7b5000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x1b000) = 0xffffbe7b5000
close(3)                                = 0
faccessat(AT_FDCWD, "/etc/ld.so.nohwcap", F_OK) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/lib/aarch64-linux-gnu/libc.so.6", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\3\0\0\0\0\0\0\0\0\3\0\267\0\1\0\0\0 \10\2\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=1349776, ...}) = 0
mmap(NULL, 1422168, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xffffbe62e000
mprotect(0xffffbe771000, 61440, PROT_NONE) = 0
mmap(0xffffbe780000, 24576, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x142000) = 0xffffbe780000
mmap(0xffffbe786000, 13144, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0xffffbe786000
close(3)                                = 0
faccessat(AT_FDCWD, "/etc/ld.so.nohwcap", F_OK) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/lib/aarch64-linux-gnu/libpthread.so.0", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\3\0\0\0\0\0\0\0\0\3\0\267\0\1\0\0\0\360\\\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=134688, ...}) = 0
mmap(NULL, 177104, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xffffbe602000
mprotect(0xffffbe619000, 61440, PROT_NONE) = 0
mmap(0xffffbe628000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x16000) = 0xffffbe628000
mmap(0xffffbe62a000, 13264, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0xffffbe62a000
close(3)                                = 0
faccessat(AT_FDCWD, "/etc/ld.so.nohwcap", F_OK) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/lib/aarch64-linux-gnu/libdl.so.2", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0\267\0\1\0\0\0\320\16\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0644, st_size=14480, ...}) = 0
mmap(NULL, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0xffffbe600000
mmap(NULL, 78080, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0xffffbe5ec000
mprotect(0xffffbe5ee000, 65536, PROT_NONE) = 0
mmap(0xffffbe5fe000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x2000) = 0xffffbe5fe000
close(3)                                = 0
mmap(NULL, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0xffffbe5ea000
mprotect(0xffffbe780000, 16384, PROT_READ) = 0
mprotect(0xffffbe5fe000, 4096, PROT_READ) = 0
mprotect(0xffffbe628000, 4096, PROT_READ) = 0
mprotect(0xffffbe7b5000, 4096, PROT_READ) = 0
mprotect(0xffffbe9f4000, 4096, PROT_READ) = 0
mprotect(0xffffbe8b5000, 12288, PROT_READ) = 0
mprotect(0xffffbe93a000, 4096, PROT_READ) = 0
mprotect(0xaaaac198e000, 4096, PROT_READ) = 0
mprotect(0xffffbea25000, 4096, PROT_READ) = 0
munmap(0xffffbea15000, 55765)           = 0
set_tid_address(0xffffbe5ea0e0)         = 2601
set_robust_list(0xffffbe5ea0f0, 24)     = 0
rt_sigaction(SIGRTMIN, {sa_handler=0xffffbe607768, sa_mask=[], sa_flags=SA_SIGINFO}, NULL, 8) = 0
rt_sigaction(SIGRT_1, {sa_handler=0xffffbe607838, sa_mask=[], sa_flags=SA_RESTART|SA_SIGINFO}, NULL, 8) = 0
rt_sigprocmask(SIG_UNBLOCK, [RTMIN RT_1], NULL, 8) = 0
prlimit64(0, RLIMIT_STACK, NULL, {rlim_cur=8192*1024, rlim_max=RLIM64_INFINITY}) = 0
brk(NULL)                               = 0xaaaadde35000
brk(0xaaaadde56000)                     = 0xaaaadde56000
mmap(NULL, 187647121166336, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = -1 ENOMEM (Cannot allocate memory)
brk(0x15554dde59000)                    = 0xaaaadde56000
mmap(NULL, 187647121297408, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = -1 ENOMEM (Cannot allocate memory)
rt_sigprocmask(SIG_UNBLOCK, [ABRT], NULL, 8) = 0
rt_sigprocmask(SIG_BLOCK, ~[RTMIN RT_1], [], 8) = 0
getpid()                                = 2601
gettid()                                = 2601
tgkill(2601, 2601, SIGABRT)             = 0
rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
--- SIGABRT {si_signo=SIGABRT, si_code=SI_TKILL, si_pid=2601, si_uid=1000} ---
+++ killed by SIGABRT +++
Aborted
```

See [full log](https://gist.github.com/jsarenik/6923be59beb524b75a154fd11b47263a)

--- a/lightningd/params.h
+++ b/lightningd/params.h
@@ -64,7 +64,7 @@ typedef bool(*param_cb)(const char *buffer, const jsmntok_t *tok, void *arg);
 		  (arg) + 0*sizeof((cb)((const char *)NULL,		\
 					(const jsmntok_t *)NULL,	\
 					(arg)) == true),		\
-		  0
+		  (size_t)0
 /*
  * Similar to above but for optional parameters.
  * @arg must be the address of a pointer. If found during parsing, it will be

